File: ios/AuraKit/Sources/AuraKit/Protocol.swift
// Protocol.swift — кадры WS-протокола Aura (те же, что у C++/Qt-клиентов).
//
// Запрос клиента:  {"id": "...", "type": "auth.login", "payload": {...}}
// Ответ сервера:   {"id": "...", "type": "ok",    "payload": {...}}
//                  {"id": "...", "type": "error", "code": "...", "message": "..."}
// Событие сервера: {"type": "event", "event": "chat.message", "payload": {...}}
import Foundation

public struct AuraRequestFrame: Encodable {
    public let id: String
    public let type: String
    public let payload: JSONValue

    public init(id: String, type: String, payload: JSONValue) {
        self.id = id
        self.type = type
        self.payload = payload
    }
}

public struct AuraResponseFrame: Decodable {
    public let id: String?
    public let type: String?
    public let code: String?
    public let message: String?
    public let event: String?
    public let payload: JSONValue?
}

/// Ошибка протокола: сервер вернул кадр {"type":"error", code, message}.
public struct AuraError: Error, LocalizedError, Equatable, Sendable {
    public let code: String
    public let message: String

    public init(code: String, message: String) {
        self.code = code
        self.message = message
    }

    public var errorDescription: String? { message }
}

/// Имена событий сервера (push-уведомления внутри WS-сессии).
public enum AuraEventName {
    public static let chatMessage = "chat.message"
    public static let taskDue = "task.due"
}

File: android/aurakit/src/main/kotlin/ai/aura/kit/Protocol.kt
package ai.aura.kit

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

// Кадры WS-протокола Aura — те же, что у C++/Qt/Swift-клиентов
// (docs/PROTOCOL.md):
//
//   запрос клиента:  {"id": "...", "type": "auth.login", "payload": {...}}
//   ответ сервера:   {"id": "...", "type": "ok",    "payload": {...}}
//                    {"id": "...", "type": "error", "code": "...", "message": "..."}
//   событие сервера: {"type": "event", "event": "chat.message", "payload": {...}}

@Serializable
data class AuraRequestFrame(
    val id: String,
    val type: String,
    val payload: JsonObject,
)

@Serializable
data class AuraResponseFrame(
    val id: String? = null,
    val type: String? = null,
    val code: String? = null,
    val message: String? = null,
    val event: String? = null,
    val payload: JsonObject? = null,
)

/** Ошибка протокола: сервер вернул кадр {"type":"error", code, message}. */
data class AuraError(val code: String, val message: String) :
    Exception("$code: $message")

/** Имена событий сервера (push внутри WS-сессии). */
object AuraEventName {
    const val CHAT_MESSAGE = "chat.message"
    const val TASK_DUE = "task.due"
    const val NOTIFICATION_NEW = "notification.new"
}
